cli/src/main/java/com/devonfw/tools/ide/commandlet/CreateCommandlet.java

```diff
@@ -58,13 +68,40 @@ public void run() {
 
     initializeProject(newProjectPath);
     this.context.setIdeHome(newProjectPath);
+    if (codeRepositoryFlag.isTrue()) {
+      String repoUrl = codeRepository.getValue();
+      if (repoUrl.isEmpty()) {
+        this.context.info("Please provide a repository URL after using --code");
+      } else {
+        initializeCodeRepository(repoUrl);
+      }
+    }
     super.run();
```

Comment:
this will still call `updateSettings()`. Is your intention to initialize the `settings` before so it will afterwards `update` the repo that has just been cloned? IMHO this is still confusing since milliseconds after cloning we do not expect changes and updating again does not make much sense.
It would IMHO make more sense to override `updateSettings()` method or to extract a new protected method out of `updateSettings()` that does the cloning part and potentially asks for settings URL. Then you can override only this method here.

Reply:
done, overrode updateSettings()

Reply:
Please push your changes. They did not yet arrive on github so it is not yet done from my PoV.

Reply:
Was a refresh problem. Sorry for confusion. Thanks for the great rework. Resolved.